### DISCUSSION TITLE
Issue while using authelia with traefik on kubernetes

### DISCUSSION THREAD

**Discussion by shinderupesh** [authelia/authelia#4121] [Q&A] [answered]
I am trying to use authelia with traefik on kubernetes. For this I am using [this](https://medium.com/@TCheronneau/sso-with-traefik-and-kubernetes-d008f9a1328a) link as a reference.

Only difference is I am using Ingress resource instead of IngressRoute. Following is my ingress yaml file:

  	---
	apiVersion: networking.k8s.io/v1
	kind: Ingress
	metadata:
	  name: tst
	  namespace: app
	  annotations:
		traefik.ingress.kubernetes.io/router.middlewares: default-authelia@kubernetescrd
		traefik.ingress.kubernetes.io/router.tls: "true"
		cert-manager.io/cluster-issuer: letsencrypt-prod
		kubernetes.io/ingress.class: traefik
	spec:
	  tls:
	  - hosts:
		- "app.dns.com"
		secretName: new-tst
	  rules:
		- host: app.dns.com
		  http:
	              paths:
		       - path: /
			    pathType: Prefix
				backend:
				  service:
					name: frontend-prod
					port:
					  number: 80

But while creating Ingress resource I am getting following issue in traefik pod of kube-system:

      level=error msg="middleware \"default-authelia@kubernetescrd\" does not exist" entryPointName=http routerName=tst-app.dns.com@kubernetes
      level=error msg="middleware \"default-authelia@kubernetescrd\" does not exist" routerName=tst-app.dns.com@kubernetes entryPointName=https

I am able to access authelia dashboard successfully. But it does not redirect to dashboard of my application.

**Comment by james-d-elliott** [comment, marked as answer]
This isn't really authelia that's the issue but show the middleware with its name and namespace.

**Reply by shinderupesh** [reply]
Here is my middleware file:

apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
    name: authelia
spec:
    forwardAuth:
        address: http://authelia-service.auth.svc.cluster.local:9091/api/verify?rd=https://auth.dns.com/
        trustForwardHeader: true
        authResponseHeaders:
            - Remote-User
            - Remote-Groups
            - Remote-Name
            - Remote-Email

**Reply by james-d-elliott** [reply]
Run `kubectl get middleware --all-namespaces`

**Reply by shinderupesh** [reply]
$ sudo kubectl get middleware --all-namespaces
NAMESPACE   NAME       AGE
default         authelia        6h

**Reply by james-d-elliott** [reply]
Show your traefik pod configuration, looking like you didn't even enable the KubernetesCRD at all. You can do this with `kubectl describe pod --namespace <traefik namespace> <traefik pod name>`. We are looking specifically for the enabled providers.

**Reply by shinderupesh** [reply]
Ok. May be that's the issue in my side. Here is my configuration of traefik pod:

	$ sudo kubectl describe pod --namespace kube-system traefik-689dcfb858-jfb2q
	Name:         traefik-689dcfb858-jfb2q
	Namespace:    kube-system
	Priority:     0
	Node:         ip-172-3-41-60/172.3.41.60
	Start Time:   Mon, 03 Oct 2022 05:38:32 +0000
	Labels:       app=traefik
				  pod-template-hash=689dcfb858
	Annotations:  <none>
	Status:       Running
	IP:           10.45.0.41
	IPs:
	  IP:           10.45.0.41
	Controlled By:  ReplicaSet/traefik-689dcfb858
	Containers:
	  traefik:
		Container ID:  containerd://7601d78874683ee9ef7358f95736436822c09638ce5c99ca970fdd415c001845
		Image:         traefik:v2.4
		Image ID:      docker.io/library/traefik@sha256:840e948af3c8d1e45e986eee7d97004ab29cfccffdf0be4c116ba9aaeff5d17a
		Ports:         80/TCP, 443/TCP, 8080/TCP
		Host Ports:    0/TCP, 0/TCP, 0/TCP
		Args:
		  --api.insecure
		  --accesslog
		  --entrypoints.https.Address=:443
		  --entrypoints.http.Address=:80
		  --providers.kubernetesIngress.ingressClass=traefik
		State:          Running
		  Started:      Tue, 04 Oct 2022 04:36:46 +0000
		Last State:     Terminated
		  Reason:       Unknown
		  Exit Code:    255
		  Started:      Mon, 03 Oct 2022 15:18:41 +0000
		  Finished:     Tue, 04 Oct 2022 04:36:21 +0000
		Ready:          True
		Restart Count:  3
		Environment:    <none>
		Mounts:
		  /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-gn7f7 (ro)
	Conditions:
	  Type              Status
	  Initialized       True
	  Ready             True
	  ContainersReady   True
	  PodScheduled      True
	Volumes:
	  kube-api-access-gn7f7:
		Type:                    Projected (a volume that contains injected data from multiple sources)
		TokenExpirationSeconds:  3607
		ConfigMapName:           kube-root-ca.crt
		ConfigMapOptional:       <nil>
		DownwardAPI:             true
	QoS Class:                   BestEffort
	Node-Selectors:              <none>
	Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
								 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
	Events:
	  Type    Reason          Age                From     Message
	  ----    ------          ----               ----     -------
	  Normal  SandboxChanged  13h                kubelet  Pod sandbox changed, it will be killed and re-created.
	  Normal  Pulled          13h                kubelet  Container image "traefik:v2.4" already present on machine
	  Normal  Created         13h                kubelet  Created container traefik
	  Normal  Started         13h                kubelet  Started container traefik

**Reply by james-d-elliott** [reply]
As suspected, it's missing `--providers.kubernetesCRD`.

> Ok. May be that's the issue in my side. 

Edit: Of course. There was never any way that error originally posted could be caused by Authelia.

**Reply by shinderupesh** [reply]
Yes, sorry for disturbing you. I used kubernetesCRD in traefik and now i am able to redirect to authelia while accessing my dashboard.
Thanks..

**Reply by james-d-elliott** [reply]
No worries at all, the traefik docs in this area are poor.. would have thought that error would have been more descriptive, glad it's solved.
